app/src/main/java/com/example/plantsapp/data/dao/RoomPlantsDao.kt

```diff
@@ -17,4 +18,10 @@ interface RoomPlantsDao {
 
     @Insert
     suspend fun insert(plant: RoomPlant)
+
+    @Delete
+    suspend fun delete(plant: RoomPlant)
+
+    @Query("DELETE FROM tasks WHERE plantName = :plantName")
+    suspend fun deleteTasksByPlantName(plantName: String)
```

Comment:
Maybe you can somehow delete tasks by CASCADE? Because you have Relation for plant to tasks?